Source/HyprMXAdapter.swift

```diff
@@ -54,18 +55,20 @@ final class HyprMXAdapter: PartnerAdapter {
             completion(error)
             return
         }
-        guard let userID = HyperMXAdapterConfiguration.userID else {
-            let error = error(.initializationFailureInvalidCredentials, description: "HyprMX reqiures a permanent userId to initialize their SDK")
-            log(.setUpFailed(error))
-            completion(error)
-            return
-        }
         initializationCompletion = completion
 
+        var gameId: String!
```

Comment:
It can definitely be gameID.


As for the optional variable, that was the only way could imagine it working if I handled the UserDefaults retrieval with an `if let`. If I use nil coalescing instead, I can't know if the value was already stored but I can just re-save it every time. Is this better?
```swift
let gameID = UserDefaults.standard.object(forKey: GAMEID_STORAGE_KEY) as? String ?? ProcessInfo.processInfo.globallyUniqueString
UserDefaults.standard.set(gameID, forKey: GAMEID_STORAGE_KEY)
```

Reply:
You could also keep your current logic:
```
let gameID: String
if let storedGameId = UserDefaults.standard.object(forKey: GAMEID_STORAGE_KEY) as? String {
            gameId = storedGameId
        } else {
            gameId = ProcessInfo.processInfo.globallyUniqueString
            UserDefaults.standard.set(gameId, forKey: GAMEID_STORAGE_KEY)
}
```

Reply:
Oh wow, you're right. I didn't think you could declare a `let` without setting it unless it was a property that gets set in an initializer. Very good to know.